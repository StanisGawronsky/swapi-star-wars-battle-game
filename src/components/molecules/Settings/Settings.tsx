import React from "react";
import { Select } from "~/components/atoms";

interface IProps {
  playerTypes: string[];
  availableAttributes: string[];
  setCurrentAttribute: (player: string) => void;
  setCurrentPlayerType: (attribute: string) => void;
}
const Settings: React.FC<IProps> = ({
  playerTypes,
  availableAttributes,
  setCurrentAttribute,
  setCurrentPlayerType,
}: IProps) => {
  return (
    <div>
      <Select
        options={playerTypes}
        label={"Chose your player type"}
        onChange={setCurrentPlayerType}
      />
      <Select
        options={availableAttributes}
        label={"Chose your attribute"}
        onChange={setCurrentAttribute}
      />
    </div>
  );
};

export default Settings;
