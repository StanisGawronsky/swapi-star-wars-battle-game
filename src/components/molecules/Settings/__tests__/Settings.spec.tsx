import { render } from "@testing-library/react";
import React from "react";
import Settings from "../Settings";

describe("<Settings />", () => {
  it("renders correctly", () => {
    const { container } = render(<Settings />);
    expect(container).toMatchSnapshot();
  });
});
