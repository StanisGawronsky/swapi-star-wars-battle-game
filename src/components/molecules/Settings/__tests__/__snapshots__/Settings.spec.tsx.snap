// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<div>
  <div>
    <div>
      <label>
        Chose your player type
      </label>
      <select />
    </div>
    <div>
      <label>
        Chose your attribute
      </label>
      <select />
    </div>
  </div>
</div>
`;
